src/main/java/co/rsk/federate/FedNodeRunner.java

```diff
@@ -202,6 +185,52 @@ private ECDSASigner buildSigner() {
         return compositeSigner;
     }
 
+    private HSMBookkeepingClient buildBookKeepingClient() {
+        SignerConfig signerConfig = this.config.signerConfig(BTC_KEY_ID.getId());
+        if ("hsm".equals(signerConfig.getType())) {
+            try {
+                bridgeConstants = this.config.getNetworkConstants().getBridgeConstants();
+                PowHSMBookkeepingConfig bookKeepingConfig = new PowHSMBookkeepingConfig(
+                        signerConfig,
+                        bridgeConstants.getBtcParamsString()
+                );
+                HSMClientProtocol protocol = HSMClientProtocolFactory.buildHSMClientProtocolFromConfig(signerConfig);
+                HSMBookKeepingClientProvider clientProvider = new HSMBookKeepingClientProvider();
+                HSMBookkeepingClient bookKeepingClient = clientProvider.getHSMBookKeepingClient(protocol);
+                bookKeepingClient.setMaxChunkSizeToHsm(bookKeepingConfig.getMaxChunkSizeToHsm());
+                LOGGER.info("[buildBookKeepingClient] HSMBookkeeping Client built for HSM version: {}", bookKeepingClient.getVersion());
+                return bookKeepingClient;
+            } catch (HSMClientException e) {
+                LOGGER.warn("[buildBookKeepingClient] Error trying to build the book keeping client {}", e.getMessage());
+            }
+        }
+        return null;
```

Comment:
Not a fan of returning `null`. We could add a check at the beggining of the method that this only works for HSM signers

```
if (!"hsm".equals(signerConfig.getType())) {
 throw exception("...)

Reply:
What do you think?

Reply:
Resolved